### DISCUSSION TITLE
Possibility to setup a time range for more than 1 year?

### DISCUSSION THREAD

**Discussion by s1lverkin** [ActivityWatch/activitywatch#1088] [Q&A]
I do know that this is possible with the query exporer, but it would be handy to be able to pull all of that data within web-ui dashboard :)

**Comment by 0xbrayo** [comment]
No it's not currently possible. But a good idea for sure, could be enabled under dev mode. Since timeouts are to be expected when dealing with that much data.

**Reply by s1lverkin** [reply]
For me with rust it loads all year data within 2 minutes, and it takes like 1.3GB of memory

**Reply by 0xbrayo** [reply]
I bet with the right optimizations, such as caching each month's query result, this can  greatly be improved on.
